homeassistant/helpers/entity_component.py

```diff
@@ -217,6 +218,40 @@ async def async_extract_from_service(
             self.hass, self.entities, service_call, expand_group
         )
 
+    @callback
+    def async_register_legacy_entity_service(
```

Comment:
I was hesitant about adding this method instead of making the two integrations handle it, but I guess it's good as it's used in two places.